src/rtcModule/webrtc.cpp

```diff
@@ -2975,13 +3056,21 @@ void Session::sendAv(AvFlags av)
 {
     cmd(RTCMD_MUTE, av.value());
 }
-Promise<void> Session::sendOffer()
+
+promise::Promise<void> Session::createRtcConnSendOffer()
 {
     assert(mIsJoiner); // the joiner sends the SDP offer
     assert(mPeerAnonId);
     createRtcConn();
+    return sendOffer();
+}
+
+Promise<void> Session::sendOffer()
+{
     auto wptr = weakHandle();
-    return mRtcConn.createOffer(pcConstraints())
+    bool firstOffer = mState < kStateInProgress;
```

Comment:
Suggestion: `bool isRenegotiation = mState == kStateInProgress;`
It seems more readable to me. I'll refactorize it.